evm/db/tracked.py

```diff
@@ -0,0 +1,84 @@
+from .backends.base import BaseDB
+
+
+class TrackedDB(BaseDB):
+    """
+    The StateDB would be responsible for collecting all the touched keys.
+
+    reads: the dict of read key-value
+    writes: the dict of written key-value
+    """
+
+    wrapped_db = None
+    _reads = None
+    _writes = None
+
+    def __init__(self, db):
+        self.wrapped_db = db
+        self._reads = {}
+        self._writes = {}
+
+    def get(self, key):
+        """
+        Return the value of specific key and update read dict.
+        """
+        try:
+            current_value = self.wrapped_db.get(key)
+        except KeyError:
+            raise
```

Comment:
This whole `try/except/else` block is equivalent to the same code without the `try/except/else` since `except` just re-raises.

```python
current_value = self.wrapped_db.get(key)
self._reads[key] = current_value
return current_value
```